### PR TITLE
Ci.cleanup

### PR DESCRIPTION
**Pull request by laughinghan**
Major changes:
- throughout circle.yml, use the `json` cli tool
- fix the Sauce Connect retry-ing to only retry on failure rather than always
- on Sauce, use human-readable build name similar to on Circle; in particular, mention the CircleCI build number
- when applicable keep around pieces of screenshots in `imgs/pieces/`, and file away the baseline imgs (formerly the `PREV_*` imgs) in `imgs/baseline/`
- screenshot filename mentions version of browser used (which changes over time for evergreen browsers)
- `script/screenshots.js` now uses the "promise chain"-style of `wd`, much fewer callbacks

Also lots of minor improvements to comments, screenshots script error logging, test names on Sauce, etc.

### CODE REVIEW

File: circle.yml [hunk 1]
```diff
@@ -31,171 +31,204 @@
 # this file is based on https://github.com/circleci/sauce-connect/blob/a65e41c91e02550ce56c75740a422bebc4acbf6f/circle.yml
 # via https://circleci.com/docs/browser-testing-with-sauce-labs/
 
+machine:
+  environment:
+    json: node_modules/.bin/json
```
Comment (desa) [inline circle.yml]: Whats this do?

File: circle.yml [hunk 2]
```diff
@@ -31,171 +31,204 @@
...
+        test -e ~/sauce_was_ready && exit
+        time sc-*-linux/bin/sc --user $SAUCE_USERNAME --api-key $SAUCE_ACCESS_KEY \
+          --readyfile ~/sauce_is_ready
+        test -e ~/sauce_was_ready && exit
+        echo 'ERROR: Exited twice without creating readyfile' \
+          | tee /dev/stderr > ~/sauce_is_ready
+        exit 1
       :
         background: true
+    - test -x $json || npm install json
```
Comment (desa) [inline circle.yml]: we might consider using `jq` instead since I think that comes with the standard Ubuntu install. Maybe not though

**Comment by desa** [comment]
I have to say reading your commits is always enjoyable.

File: circle.yml (hunk 2 above)
Comment (laughinghan) [inline circle.yml]: `jq` is fucking dope!

Fixed by https://github.com/mathquill/mathquill/pull/723/commits/50abeb4a2e2486b8f7bd8a6a5ebbc3125b8dea1c

Comment (laughinghan) [inline circle.yml]: how do you know all the cool JSON utilities?!

**Pull request merged**